src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -24,35 +24,118 @@ import DataIntegration from '../../../../components/data-processing/DataIntegrat
 import ConfigureEmbedding from '../../../../components/data-processing/ConfigureEmbedding/ConfigureEmbedding';
 import { completeProcessingStep, loadProcessingSettings } from '../../../../redux/actions/experimentSettings';
 
+import SingleComponentMultipleDataContainer from '../../../../components/SingleComponentMultipleDataContainer';
+
 const { Text } = Typography;
 const { Option } = Select;
 
 const DataProcessingPage = ({ experimentId, experimentData, route }) => {
+  const cellSetsProperties = useSelector((state) => state.cellSets.properties);
+  const cellSetsHierarchy = useSelector((state) => state.cellSets.hierarchy);
+
+  const sampleKeys = cellSetsHierarchy.find(
+    (rootNode) => (rootNode.key === 'sample'),
+  ).children.map(
```

Comment:
This will fail if the cellsets are not loaded. We should check if the cellsets are there and load them if not.